sdk/src/fee.rs

```diff
@@ -122,13 +135,14 @@ impl FeeStructure {
                     .unwrap_or_default()
             });
 
-        ((budget_limits
-            .prioritization_fee
-            .saturating_add(signature_fee)
-            .saturating_add(write_lock_fee)
-            .saturating_add(compute_fee) as f64)
-            * congestion_multiplier)
-            .round() as u64
+        FeeDetails {
+            transaction_fee: ((signature_fee
+                .saturating_add(write_lock_fee)
+                .saturating_add(compute_fee) as f64)
+                * congestion_multiplier)
+                .round() as u64,
+            prioritization_fee: budget_limits.prioritization_fee,
+        }
```

Comment:
I think this is slightly different. If the congestion multiplier is not 1.0, then we could end up with a different total fee.

We had:
$F = \text{round}(\text{Con} * (P + S + W + C))$

but now we have:

$F = P + \text{round}(\text{Con} * (S +W +C))$

where $F$ is total fee, $\text{Con}$ is congestion multiplier, $P$ is priority fee, $S$ is signature fee, $W$ is write lock fee, and $C$ is compute fee.

I think congestion multiplier will always be 1 on MNB? is it not 1 anywhere? I think you have a PR to remove this anyway...if that's the case I'd probably feel more comfortable merging this change in **after** that PR. wdyt?

Reply:
congestion_multiplier is always 1, except in tests it could be 0.

but yea, let's land #34537 first.

Reply:
rebased.